Tests/Core/AWSClientRuntimeTests/Protocols/Ec2Query/Ec2ErrorRequestIdTests.swift

```diff
@@ -40,34 +42,37 @@ class Ec2ErrorRequestIdTests: XCTestCase {
             </Errors>
             <RequestId>abcdefg12345</RequestId>
         </Ec2Response>
-        """.data(using: .utf8)!
-        let response = try XMLDecoder().decode(Ec2Response.self, from: data)
+        """.utf8)
+        let httpResponse = HttpResponse(body: .data(data), statusCode: .ok)
+        let response = try await responseClosure(Ec2Response.httpBinding, responseDocumentBinding)(httpResponse)
         XCTAssertEqual(response.requestId, "abcdefg12345")
     }
 
-    func testEc2NarrowedResponseDecodesRequestID() throws {
-        let data = """
+    func testEc2NarrowedResponseDecodesRequestID() async throws {
+        let data = Data("""
         <Ec2NarrowedResponse>
             <Errors>
                 <Error>Sample Error</Error>
             </Errors>
             <RequestID>abcdefg12345</RequestID>
         </Ec2NarrowedResponse>
-        """.data(using: .utf8)!
-        let response = try XMLDecoder().decode(Ec2NarrowedResponse<String>.self, from: data)
+        """.utf8)
+        let httpResponse = HttpResponse(body: .data(data), statusCode: .ok)
+        let response = try await responseClosure(Ec2Response.httpBinding, responseDocumentBinding)(httpResponse)
         XCTAssertEqual(response.requestId, "abcdefg12345")
     }
 
-    func testEc2NarrowResponseDecodesRequestId() throws {
-        let data = """
+    func testEc2NarrowResponseDecodesRequestId() async throws {
+        let data = Data("""
         <Ec2Response>
             <Errors>
                 <Error>Sample Error</Error>
             </Errors>
             <RequestId>abcdefg12345</RequestId>
         </Ec2Response>
-        """.data(using: .utf8)!
-        let response = try XMLDecoder().decode(Ec2NarrowedResponse<String>.self, from: data)
+        """.utf8)
+        let httpResponse = HttpResponse(body: .data(data), statusCode: .ok)
+        let response = try await responseClosure(Ec2Response.httpBinding, responseDocumentBinding)(httpResponse)
```

Comment:
The tests for EC2 errors above are converted to use closures instead of decoders to create the errors.

The tests for the several following deleted files were removed because they appear to have been copied from protocol tests.  I suspect they were used in the early days of establishing protocol tests in this project.

The same tests still exist in the protocol tests suite, and they are being removed here as redundant.